Modules/_ctypes/callproc.c

```diff
@@ -835,15 +836,33 @@ static int _call_function_pointer(int flags,
     if ((flags & FUNCFLAG_CDECL) == 0)
         cc = FFI_STDCALL;
 #endif
-    if (FFI_OK != ffi_prep_cif(&cif,
-                               cc,
-                               argcount,
-                               restype,
-                               atypes)) {
-        PyErr_SetString(PyExc_RuntimeError,
-                        "ffi_prep_cif failed");
-        return -1;
+
+#if defined(__APPLE__) && __arm64__
```

Comment:
I'm not sure about this. This is the least impacting change to get a working ctypes on macOS/arm64, but this complicates the code base.

For the trunk I'd prefer to use ffi_pre_cif_var unconditionally when available instead of only on macOS/arm64.

Reply:
how can we determine if it is available?   We do not decide which libffi to use until `setup.py` gets run, so we can't use autoconf to prove for it.     Should we add a autoconf style probe in `setup.py`?

Reply:
OK, if you like the idea of setup.py probing for functions in `libffi`, take a look here: https://github.com/python/cpython/pull/21249

I thought to keep things simple it would be best to just grep the headers, rather than to try to do what autoconf would have done.